fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/DistributionSpec.java

```diff
@@ -27,6 +27,9 @@ public static class PropertyInfo {
         public List<Long> partitionIds = Lists.newArrayList();
         // record nullable columns generated by outer join
         public ColumnRefSet nullableColumns = new ColumnRefSet();
+        // join on columns, cannot use ColumnRefSet because order matters
+        public List<Integer> joinOnLeftColumns = Lists.newArrayList();
+        public List<Integer> joinOnRightColumns = Lists.newArrayList();
```

Comment:
Maybe you should consider more level+ join.
like:
``` 
A  JOIN[bucket] B on A.a = B.b
    JOIN[bucket] C on B.b = C.c
    JOIN[bucket] D on C.c = D.d
    JOIN[bucket] E on D.d = E.e
```

Reply:
And why need split left columns and right columns?
```
select A.a, sum(1)
from (select * from B join[bucket] A on B.b = A.a) 
group by A.a
```
it's canbe work?

Reply:
Thanks a lot for giving advise.

It has been updated, and not it can handle multi-level join. And I have replaced left columns and right columns with a unified ColumnRefSet.

「select A.a, sum(1)
from (select * from B join[bucket] A on B.b = A.a) 
group by A.a」
this case works fine, and both case you memtioned was added to ut